### PR TITLE
Bug/33598 edit group with trailing space in name

### PR DESCRIPTION
**Pull request by danwilliams**

### CODE REVIEW

File: admin/services/group.service.js [hunk 1]
```diff
@@ -83,6 +83,9 @@ groupService.update = async (id, group, schoolId) => {
   if (!(pupils instanceof Array)) {
     pupils = Object.values(pupils)
   }
+  if (typeof group.name === 'string') {
```
Comment (GeorgeChatzigiannis) [inline admin/services/group.service.js]: Is there a case where this could come as a non-string value?

File: admin/services/group.service.js [hunk 2]
```diff
@@ -104,6 +107,9 @@ groupService.create = async (groupName, groupPupils, schoolId) => {
   if (!groupName || !schoolId) {
     throw new Error('groupName and schoolId are required')
   }
+  if (typeof groupName === 'string') {
```
Comment (GeorgeChatzigiannis) [inline admin/services/group.service.js]: Same as my other question

File: admin/services/group.service.js (hunk 1 above)
Comment (danwilliams) [reply admin/services/group.service.js]: Not really possible, since in both cases we're using a validator that name exists and is a string.

Comment (GeorgeChatzigiannis) [reply admin/services/group.service.js]: @danwilliams wouldn't that make the if condition redundant then?

Comment (GeorgeChatzigiannis) [reply admin/services/group.service.js]: Resolving since a newer commit covered that question

**Pull request merged**